e2e/client/types.go

```diff
@@ -230,14 +230,14 @@ func podMeta(podName string, namespace string, label map[string]string, annotati
 }
 
 func dynamicNetworksAnnotation(pod *corev1.Pod, newIfaceConfigs ...*nettypes.NetworkSelectionElement) string {
+	var currentNetworkSelectionElements []*nettypes.NetworkSelectionElement
 	currentNetworkSelectionElementsString, wasFound := pod.ObjectMeta.Annotations[nettypes.NetworkAttachmentAnnot]
-	if !wasFound {
-		return ""
-	}
-
-	currentNetworkSelectionElements, err := annotations.ParsePodNetworkAnnotations(currentNetworkSelectionElementsString, pod.GetNamespace())
-	if err != nil {
-		return ""
+	if wasFound {
+		var err error
+		currentNetworkSelectionElements, err = annotations.ParsePodNetworkAnnotations(currentNetworkSelectionElementsString, pod.GetNamespace())
+		if err != nil {
+			return ""
+		}
```

Comment:
Could you extract this part to its own function? I think that the logic looking up and parsing the annotation (if any) is bloating this function (where I would like to read "get elements, update elements, marshal elements").

Reply:
Done.